### PR TITLE
ReportTemplate Nailgun E2E test

### PR DESCRIPTION
**Pull request by lhellebr**

### CODE REVIEW

**Comment by lhellebr** [comment]
Depends on https://github.com/SatelliteQE/nailgun/pull/629

File: tests/foreman/api/test_reporttemplates.py [hunk 1]
```diff
@@ -16,102 +16,56 @@
 :Upstream: No
 """
 
+from robottelo.datafactory import gen_string
 from robottelo.decorators import tier1, tier2, stubbed
 from robottelo.test import APITestCase
 
+from nailgun import entities
```
Comment (vijay8451) [inline tests/foreman/api/test_reporttemplates.py]: can we set in alphabetical order please ..

File: tests/foreman/api/test_reporttemplates.py [hunk 2]
```diff
@@ -16,102 +16,56 @@
...
         """
+        name = gen_string('alpha')
+        new_name = gen_string('alpha')
+        template = gen_string('alpha')
+        # create and read report template
+        entities.ReportTemplate(name=name, template=template).create()
+        rt = entities.ReportTemplate().search(query={'search': 'name={0}'.format(name)})[0].read()
+        self.assertEquals(name, rt.name)
+        self.assertEquals(template, rt.template)
+        # update report template
```
Comment (vijay8451) [inline tests/foreman/api/test_reporttemplates.py]: Can you try to modify any other field than 'name' i.e. unlock or 'Template input' ... what do you think ?

File: tests/foreman/api/test_reporttemplates.py [hunk 3]
```diff
@@ -16,102 +16,56 @@
...
 
 class ComputeResourceTestCase(APITestCase):
     """Tests for ``katello/api/v2/report_templates``."""
 
     @tier1
-    @stubbed()
-    def test_positive_create_report(self):
-        """Create report template
+    def test_positive_e2e_crud(self):
+        """Run CRUD tests
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_reporttemplates.py]: Can you please add more information in docstrings
e.g. `Execute/Run Crud operations/test for report_templates `

File: tests/foreman/api/test_reporttemplates.py [hunk 4]
```diff
@@ -16,102 +16,56 @@
...
 
         :expectedresults: Report is deleted
 
         :CaseImportance: High
         """
+        name = gen_string('alpha')
+        new_name = gen_string('alpha')
+        template = gen_string('alpha')
+        # create and read report template
+        entities.ReportTemplate(name=name, template=template).create()
```
Comment (oshtaier) [inline tests/foreman/api/test_reporttemplates.py]: You do not need to search to get entity. `Create()` always do `read()`
`rt = entities.ReportTemplate(name=name, template=template).create()`

File: tests/foreman/api/test_reporttemplates.py [hunk 5]
```diff
@@ -16,102 +16,56 @@
...
+        rt = entities.ReportTemplate().search(query={'search': 'name={0}'.format(name)})[0].read()
+        self.assertEquals(name, rt.name)
+        self.assertEquals(template, rt.template)
+        # update report template
+        entities.ReportTemplate(id=rt.id, name=new_name).update({'name'})
```
Comment (oshtaier) [inline tests/foreman/api/test_reporttemplates.py]: same with update
`rt = entities.ReportTemplate(id=rt.id, name=new_name).update({'name'})`

File: tests/foreman/api/test_reporttemplates.py [hunk 6]
```diff
@@ -16,102 +16,56 @@
...
+        rt = entities.ReportTemplate().\
+            search(query={'search': 'name={0}'.format(new_name)})[0].read()
+        self.assertEquals(new_name, rt.name)
+        self.assertEquals(template, rt.template)
+        rts = entities.ReportTemplate().search(query={'search': 'name={0}'.format(name)})
```
Comment (oshtaier) [inline tests/foreman/api/test_reporttemplates.py]:
```
        rts = entities.ReportTemplate().search(query={'search': 'name={0}'.format(new_name)})
        self.assertEqual(len(rts), 1)
```

**Comment by jyejare** [comment]
@lhellebr , Would you like to revisit and fix some of these comments ?

**Comment by lhellebr** [comment]
Actually, this PR got obsoleted by another PR of mine, https://github.com/SatelliteQE/robottelo/pull/7437 . This can be closed.

**Pull request closed without merging**